### PR TITLE
Model::findUnits bug

### PR DESCRIPTION
**Pull request by kerimoyle**
Addresses #772 and #719

### CODE REVIEW

**Comment by kerimoyle** [comment]
I'm struggling to understand two of the python tests:

in Model.test_replace_units:
```
        # bool replaceUnits(const UnitsPtr &oldUnits, const UnitsPtr &newUnits)
        m = Model()
        a = Units()
        m.addUnits(a)
        b = Units()
        self.assertFalse(m.replaceUnits(b, a))
        self.assertFalse(m.replaceUnits(a, b))   # KRM I don't know why this is expected to fail??
        self.assertFalse(m.replaceUnits(b, a))
        del [m, a, b]
```
in Model.test_remove_units:
```
        # bool removeUnits(const UnitsPtr &units)
        m = Model()
        u1 = Units()
        u2 = Units()
        self.assertFalse(m.removeUnits(u1))
        m.addUnits(u1)
        self.assertFalse(m.removeUnits(u2))
        self.assertFalse(m.removeUnits(u1)) # KRM ... why is this not expected to work??
        del [m, u1, u2]
```

**Comment by kerimoyle** [comment]
Ready for review please (assuming build passes) @agarny @hsorby @nickerso

**Comment by hsorby** [comment]
I have two tests I would like to add to this but I can't. Is the branch is locked?

**Comment by kerimoyle** [comment]
Should be unlocked now?

File: tests/model/model.cpp [hunk 1]
```diff
@@ -669,3 +669,32 @@ TEST(Model, removeImportedUnitsByNameOrReference)
...
+
+TEST(Units, removeUnitsObjectEquivalent)
+{
+    libcellml::ModelPtr m = libcellml::Model::create();
+
+    libcellml::UnitsPtr u1 = libcellml::Units::create();
+    libcellml::UnitsPtr u2 = libcellml::Units::create();
+
+    EXPECT_TRUE(libcellml::Units::equivalent(u1, u2));
+    m->addUnits(u1);
```
Comment (kerimoyle) [inline tests/model/model.cpp:694]: Comparing two base units without names that are not in the model.  There's no way that `equivalent` should be `true` when there's no information about either??

File: tests/model/model.cpp [hunk 2]
```diff
@@ -669,3 +669,32 @@ TEST(Model, removeImportedUnitsByNameOrReference)
...
+
+    EXPECT_EQ(size_t(1), m->unitsCount());
+
+    m->removeUnits(u2);
+    EXPECT_EQ(size_t(0), m->unitsCount());
```
Comment (kerimoyle) [inline tests/model/model.cpp:699]: I really don't like this behaviour!  When removing units by reference, it should be *only* the reference that is tested.  When removing units by name, we take the first one that matches ... just as we do for every other item type ... equivalence to other units is irrelevant.

**Comment by hsorby** [comment]
Closing this as it has been incorporated through PR #785.

**Pull request closed without merging**